### PR TITLE
E2E Tests Sink

### PR DESCRIPTION
**Pull request by muralov**
Implement a sink for E2E tests that receive and store events for later check

* Write the logic
* Create Github Action for this

#77

### CODE REVIEW

File: e2e/sink/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# the GOARCH has not a default value to allow the binary be built according to the host where the command
+# was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
+# the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
+# by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
+RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o sink cmd/main.go
+
+# Use distroless as minimal base image to package the sink binary
+# Refer to https://github.com/GoogleContainerTools/distroless for more details
+FROM gcr.io/distroless/static:nonroot
+LABEL source = git@github.com:kyma-project/eventing-manager.git
```
Comment (mfaizanse) [inline e2e/sink/Dockerfile:30]: Should we change the label here?

File: e2e/sink/cmd/main.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "8080" // default port
+	}
+
+	sHandler := handler.NewSinkHandler(logger)
+	err = sHandler.Start(port)
+	if err != nil {
+		logger.Error("failed to start SinkHandler", zap.Error(err))
```
Comment (mfaizanse) [inline e2e/sink/cmd/main.go:25]: Should we do `logger.Fatal` here? so it exits with code 1?

File: e2e/sink/internal/handler/handler.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+			h.namedLogger().Error("failed to write response", zap.Error(e))
+		}
+		return
+	}
+
+	// store the event in memory
+	h.events[event.ID()] = &sinkEvent{
+		Header: http.Header{
+			SampledHeader:     r.Header[SampledHeader],
+			TraceparentHeader: r.Header[TraceparentHeader],
```
Comment (mfaizanse) [inline e2e/sink/internal/handler/handler.go:66]: Question: Will it fail here or return empty string if these headers are missing in the request?

Comment (muralov) [reply e2e/sink/internal/handler/handler.go]: no, it just doesn't set it. I've tested it.

File: e2e/sink/Dockerfile (hunk 1 above)
Comment (muralov) [reply e2e/sink/Dockerfile]: I understood the source here as the git link to the repository.

**Review by mfaizanse — APPROVED** [review]

**Pull request merged**